src/main/java/com/systemmeltdown/meltdownlog/LogSession.java

```diff
@@ -0,0 +1,90 @@
+package com.systemmeltdown.meltdownlog;
+
+import com.systemmeltdown.meltdownlog.lib.NanoTimeReference;
+import com.systemmeltdown.meltdownlog.outputs.LogOutput;
+import com.systemmeltdown.meltdownlog.topics.LogTopic;
+import com.systemmeltdown.meltdownlog.topics.LogTopicRegistry;
+
+/**
+ * Represents a time-based session of logging. Each session is pre-configured
+ * and logs to its outputs after it's started until it is stoppped.
+ */
+public abstract class LogSession implements NanoTimeReference {
+	protected final LogTopicRegistry m_topicRegistry;
+	protected long m_startNanos = Long.MIN_VALUE;
+	protected long m_stopNanos = Long.MIN_VALUE;
+
+	public LogSession(final LogTopicRegistry topicRegistry) {
+		m_topicRegistry = topicRegistry;
+	}
+
+	public boolean addTopicOutput(final LogOutput output, final String topicName) {
+		return addTopicOutput(output, topicName, System.nanoTime());
+	}
+
+	protected boolean addTopicOutput(final LogOutput output, final String topicName, final long nanos) {
+		final LogTopic topic = m_topicRegistry.getTopic(topicName);
+
+		if (topic == null) {
+			System.err.println("LogSession.addTopicOutput: topic by name '" + topicName + "' does not exist.");
```

Comment:
How do we guarantee that all topics are created before the session is initialized? It seems better to save these and do all the subscriptions at the start call.

Reply:
Well, that would require more code for the session to hold a dynamic mapping of all topics to all sessions. I'm not sure if it's worth the extra effort or overhead.